### DISCUSSION TITLE
Welcome to ESP32Radio-V2 Discussions!

### DISCUSSION THREAD

**Discussion by Edzelf** [Edzelf/ESP32Radio-V2#114] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Edzelf** [comment]
On request, the discussion is open!

**Comment by z34pec** [comment]
Thank you. I think this is a good place to post solutions to our problems also.

**Comment by superpat999** [comment]
Hi 

 I built your original ESP32Radio on a Naked Nnija PCB. I  haven't used it for some time, it was working when I put it away,  and would like to upgrade to your new  V2.

Please can you tell me what I need to do to update my V1 to a V2?      I understand the software updates, but will the existing V1 hardware  and  circuit work with V2 software?   Do I have to modify the old pcb?

Many thanks in advance

regards       Patrick            email:- patrick@byeslane.org.uk

**Comment by Edzelf** [comment]
V2 will run on the same hardware.  Be sure to edit the config.h file.

**Comment by chainsawdave** [comment]
Hi Edzelf I have been watching your project for a while. I decided to build it. Unfortunate, I'm a noob to ESP323 and programming. I have loaded the files into platform IO and flashed the ESP32. However, it does not connect to a station and I get the following error.
"
[ 14741][I][ESP32 Radio.cpp:4372] gettime(): [main] Sync TOD
[ 14743][I][ESP32 Radio.cpp:4385] gettime(): [main] TOD synced
[ 14849][E][ESP32 Radio.cpp:749] nvsgetstr(): [main] nvs_get_str failed 1102 for key preset, keylen is 6, len is 150!
[ 14850][E][ESP32 Radio.cpp:750] nvsgetstr(): [main] Contents: 
[ 14855][I][ESP32 Radio.cpp:966] nextPreset(): [main] nextPreset is 0
[ 14861][I][ESP32 Radio.cpp:3403] radiofuncs(): [main] Radiofuncs cmd is 1
[ 14868][I][ESP32 Radio.cpp:1400] connecttohost(): [main] Connect to host
[ 14874][E][AsyncTCP.cpp:729] connect(): error: -16
[ 14878][E][ESP32 Radio.cpp:1472] connecttohost(): [main] Request  failed!
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
"
I'm getting an IP address and the webserver starts. I'm not using a screen or the rotary control yet.

What am I doing wrong?
Thank you in advance.
Cheers
David

PS Using Windows 11, ESP32 DEVKITV1, V1053

**Comment by Edzelf** [comment]
Once you can access the webinterface, set your preferences using the config page (chapter 2.2 on the documentation).

**Reply by chainsawdave** [reply]
Thank you for your rapid response. I'm unable to access the webinterface. I have an IP address but it's not loading the index file. All I get is "sorry" in my browser. Could this be the issue? "nvs_get_str failed 1102 for key preset, keylen is 6, len is 150!"

I saw this in the code too. Maybe you can tell what I'm doing wrong from this.


Core  1 register dump:
PC      : 0x400ed9a7  PS      : 0x00060d30  A0      : 0x800d885c  A1      : 0x3ffceea0
A2      : 0x00000000  A3      : 0x00000000  A4      : 0xbe10f9e1  A5      : 0x3ffcee80
A6      : 0x3ffcee80  A7      : 0x00000080  A8      : 0x800d6460  A9      : 0x3ffc6074  
A10     : 0x3ffcec38  A11     : 0x00000004  A12     : 0x3f400124  A13     : 0x000005c0
A14     : 0x3f401e54  A15     : 0x3f401ef3  SAR     : 0x0000001a  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000009  LBEG    : 0x4008bb95  LEND    : 0x4008bba5  LCOUNT  : 0xffffffff


Backtrace: 0x400ed9a4:0x3ffceea0 0x400d8859:0x3ffceec0 0x400da0b4:0x3ffceee0 0x400f1935:0x3ffcef00

  #0  0x400ed9a4:0x3ffceea0 in VS1053::setVolume(unsigned char) at lib/codecs/src/VS1053.cpp:216
  #1  0x400d8859:0x3ffceec0 in spfuncs() at src/ESP32 Radio.cpp:3352
      (inlined by) spfuncs() at src/ESP32 Radio.cpp:3327
  #2  0x400da0b4:0x3ffceee0 in loop() at src/ESP32 Radio.cpp:3468
  #3  0x400f1935:0x3ffcef00 in loopTask(void*) at C:/Users/mrdav/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:50




ELF file SHA256: 43e5400485cc23bd

E (16200) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ets Jul 29 2019 12:21:46

**Reply by Edzelf** [reply]
You have to upload the files in the "data" directory to SPIFFS.

**Reply by chainsawdave** [reply]
Thank you so much Ed!! I uploaded data files using the platform in Platform IO. Initially, the webpages came up erratically and slowly. I don't know what happened, when i came back to it this morning. The web interface was working well. Nice job!
I added the rotary encoder. Can it run on 3.3Vfrom the ESP32?  Also, have you tried taking 5V off the ESP32 from the VIN input when it's on USB power?

**Reply by Edzelf** [reply]
Normally, the rotary encoder does not need any power.  The ESP32 inputs will supply the pull-ups.  If you have a roratry encoder with internal pull-ups, you have to connect these to 3.3 Volt.
Vin is an input for 5Volt power supply.  You cannot use it for output.

**Reply by chainsawdave** [reply]
Merry Christmas Ed. Quick update. Web interface working as intended. My rotary encoded seems to be able to change stations but as you noted requires 3.3V. I'm seeing the station connecting and I think it's streaming however the VS1053 is not putting out anything other than regular clicks and pops.

The board I'm using says it's a VS1053/VS1003. Not sure why the software says it's a VS1001. 

I pulled this error.
 25001][I][main.cpp:1400] connecttohost(): [main] Connect to host airspectrum.cdnstream1.com:8114/1648_128
[ 25026][I][VS1053.cpp:129] testComm(): [VS1053] Slow SPI, Testing VS1053 read/write registers...
[ 25029][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:012C R1:0000 R2:0000
[ 25042][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0258 R1:0000 R2:0000
[ 25053][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0384 R1:0000 R2:0000
[ 25063][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:04B0 R1:0000 R2:0000
[ 25074][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:05DC R1:0000 R2:0000
[ 25085][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0708 R1:0000 R2:0000
[ 25095][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0834 R1:0000 R2:0000
[ 25106][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0960 R1:0000 R2:0000
[ 25116][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0A8C R1:0000 R2:0000
[ 25126][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0BB8 R1:0000 R2:0000
[ 25131][I][main.cpp:2541] onConnect(): [main] Connected to host at 144.217.252.222 on port 8114
[ 25136][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0CE4 R1:0000 R2:0000
[ 25148][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0E10 R1:0000 R2:0000
[ 25158][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:0F3C R1:0000 R2:0000
[ 25168][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:1068 R1:0000 R2:0000
[ 25178][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:1194 R1:0000 R2:0000
[ 25188][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:12C0 R1:0000 R2:0000
[ 25198][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:13EC R1:0000 R2:0000
[ 25208][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:1518 R1:0000 R2:0000
[ 25211][I][main.cpp:1461] connecttohost(): [main] send GET command
[ 25218][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:1644 R1:0000 R2:0000
[ 25229][E][VS1053.cpp:150] testComm(): [VS1053] SPI error. SB:1770 R1:0000 R2:0000
[ 25240][W][VS1053.cpp:162] testComm(): [VS1053] This is not a VS1053, but a VS1001 instead!

**Comment by SP5MI** [comment]
Hi. I have some issues with my ESP. Radio chops playback, but I cant see any error. When I look at the Test I have something like this: "Free memory is 27636/80732, chunks in queue 593, bitrate 3420614 kbps", chunks in queue never goes down to 0, free memory is always on similar level. I use I2S PCM5102 module. I was trying to increase "QSIZ" from 400 to 600, but with no effect. Is this caused by my network connection? Or stream quality? It's look like this: https://www.youtube.com/watch?v=XtP9HAWWrRY. Stream is from here: audio.radiownet.pl:8000/stream64. 
Anyway, thank you Edzelf for this great project. Now, thanks to you, internet radio plays every day in my kitchen :)

**Comment by aly-fly** [comment]
I changed QSIZ to 1800 and now most of the stations play fine. Even the more remote ones. This induces noticeable latency, also when switching stations. But I prefer better playback over latency.

Reference: https://github.com/Edzelf/ESP32Radio-V2/issues/75

**Comment by technoplusnl** [comment]
I have a strange problem, when I connect the encoder the processor won't boot, if I plug it in after boot it works.
What could be the problem?
Thanks, Albert

**Comment by chainsawdave** [comment]
Hi I’ve seen this. In my case I had the wired connections 180 degrees wrong. Ground was clk etc. Ed is the best to respond. Good luck. It’s an awesome project.Cheers DavidSent from my iPhoneOn Jan 18, 2025, at 9:40 AM, technoplusnl ***@***.***> wrote:﻿
I have a strange problem, when I connect the encoder the processor won't boot, if I plug it in after boot it works.
What could be the problem?
Thanks, Albert

**Reply by technoplusnl** [reply]
Thanks for your reply David! I purchased an encoder already mounted on a small PCB including pull up resistors, the connections seem to be exactly like on the ESP32radio-V2 PCB. The encoder is fully functional once plugged in after boot... Cheers, Albert

**Comment by chainsawdave** [comment]
Awesome 👏 That’s what I’m using. I seem to remember Ed saying something about the pull up resistors can sometimes interfering with flashing the board so something just to bear in mind for the future. Easily solved.CheersDavidSent from my iPhoneOn Jan 19, 2025, at 4:31 AM, technoplusnl ***@***.***> wrote:﻿
Thanks for your reply David! I purchased an encoder already mounted on a small PCB including pull up resistors, the connections seem to be exactly like on the ESP32radio-V2 PCB. The encoder is fully functional once plugged in after boot... Cheers, Albert

**Comment by marik2500** [comment]
I managed to upload this wonderful project that I have been following for several years, on an Arduino Nano Esp32 board that is based on the ESP32S3 uC, but it does not have SPIFFS and a partition must be created on the flash. It does not work with the Arduino IDE but I managed to use PlatformIO with writing using upload protocol = esptool. Putting the Arduino Nano Esp32 board in this mode is done by shorting B1 to GND and pressing the reset button. I used the MAX98357 as a decoder because it also works with some AAC encoded radio stations. Thanks to the author!!!

**Reply by Edzelf** [reply]
Thank you for your kind words.  Have fun with the radio!

**Comment by chainsawdave** [comment]
I echo the comments. It’s a great project and Ed you been so kind with your support.Cheers DavidSent from my iPhoneOn Jul 18, 2025, at 3:55 AM, Ed Smallenburg ***@***.***> wrote:﻿
Thank you for your kind words.  Have fun with the radio!
